packages/components/src/utils/getParsedDate.ts

```diff
@@ -0,0 +1,39 @@
+import { isMatch, parse } from "date-fns"
+
+const SUPPORTED_DATE_FORMATS = [
+  "dd/MM/yyyy",
+  "d MMM yyyy",
+  "d MMMM yyyy",
+  "dd MMM yyyy",
+  "dd MMMM yyyy",
+  "yyyy-MM-dd",
+  "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+]
+
+export const getParsedDate = (dateString: string) => {
+  try {
+    const parsedDate = SUPPORTED_DATE_FORMATS.reduce<Date | undefined>(
+      (acc, format) => {
+        if (acc) {
+          // Date has already been parsed by an earlier format
+          return acc
+        }
+
+        if (isMatch(dateString, format)) {
+          return parse(dateString, format, new Date())
+        }
```

Comment:
nit: we should shift the `try/catch` here since this is the only thing that can fail

Reply:
Done in 715e22c6828872fb0861f310c808324f983f2993.